pkg/client/connector/command.go

```diff
@@ -286,9 +280,9 @@ func (s *service) connectWorker(c context.Context, cr *rpc.ConnectRequest, k8sCo
 
 	ret := &rpc.ConnectInfo{
 		Error:          rpc.ConnectInfo_UNSPECIFIED,
-		ClusterContext: s.cluster.K8sConfig.Context,
-		ClusterServer:  s.cluster.K8sConfig.Server,
-		ClusterId:      s.cluster.GetClusterId(c),
+		ClusterContext: s.sharedState.GetClusterNonBlocking().K8sConfig.Context,
+		ClusterServer:  s.sharedState.GetClusterNonBlocking().K8sConfig.Server,
+		ClusterId:      s.sharedState.GetClusterNonBlocking().GetClusterId(c),
```

Comment:
Please assign the result of `GetClusterNonBlocking()` to a variable at line 221 instead of calling it repeatedly 6 times.